core/domain/user_services.py

```diff
@@ -850,19 +1050,41 @@ def create_new_user(gae_id, email):
     Raises:
         Exception. If a user with the given gae_id already exists.
     """
-    user_settings = get_user_settings(gae_id, strict=False)
+    user_settings = get_user_settings_by_gae_id(gae_id, strict=False)
     if user_settings is not None:
-        raise Exception('User %s already exists.' % gae_id)
+        raise Exception(
+            'User %s already exists for gae_id %s.'
+            % (user_settings.user_id, gae_id))
 
     user_id = user_models.UserSettingsModel.get_new_id('')
+    # TODO(#10178): Remove gae_id attribute from UserSettings below
+    # when feconf.ENABLE_USER_AUTH_MODEL flag has been set True.
     user_settings = UserSettings(
         user_id, gae_id, email, feconf.ROLE_ID_EXPLORATION_EDITOR,
         preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE])
+
+    if feconf.ENABLE_USER_AUTH_MODEL is True:
```

Comment:
This need not be behind a flag. We can create the user auth models even before the migration is done